apps/aechannel/src/aesc_fsm.erl

```diff
@@ -1551,7 +1547,8 @@ handle_call_(open, {upd_create_contract, Opts}, From, #data{} = D) ->
     end,
     new_contract_tx_for_signing(Opts#{owner => FromPub}, From, D);
 handle_call_(open, {upd_call_contract, Opts, ExecType}, From,
-             #data{state=State, opts = ChannelOpts} = D) ->
+             #data{state=State, opts = ChannelOpts,
+                   on_chain_id = ChannelId} = D) ->
```

Comment:
The code path seems to expect a binary rather than `undefined` - could this clause be made more specific?
```suggestion
                   on_chain_id = ChannelId} = D) when ChannelId =/= undefined ->
```
Reading it is not immediate that this is not undefined - maybe for a previous operation/check?

There may be other points where this applies.